### PR TITLE
Add: Dashboard Missing Ep+Tmdb Link Shortcuts / Collection Tag Links and Sticky Searchbar+Sidebars

### PR DESCRIPTION
**Pull request by natyusha**
Also:
- added hover to the the unrecognized utility page tabs and timeline sidebar posters
- added a tooltip for the collection view edit button and hover to tag links
- made the list view posters slightly larger (to fix an alignment issue)
- add hover transitions across the webui
- slightly reduce top nav padding

should cover this issue: https://github.com/ShokoAnime/Shoko-WebUI/issues/1084

### CODE REVIEW

File: src/components/Collection/Filter/FilterSidebar.tsx [hunk 1]
```diff
@@ -70,7 +70,7 @@ const FilterSidebar = () => {
   return (
     <ShokoPanel
       title="Filter"
-      className="ml-8 w-full"
+      className="sticky top-24 ml-6 !h-[calc(100vh-18rem)] w-full"
```
Comment (harshithmohan) [inline src/components/Collection/Filter/FilterSidebar.tsx:73]: `!h-[calc(100vh-18rem)]`, this, again, as used in various other places in the codebase, is a hack.
But I will let it go since we do it elsewhere

File: src/pages/dashboard/panels/CollectionStats.tsx [hunk 2]
```diff
@@ -1,23 +1,50 @@
...
+          >
+            {value}
+          </div>
+        )
+        : <div>{value}</div>}
     </div>
-    {link ? <Link to={link} className="font-semibold text-panel-text-primary">{value}</Link> : <div>{value}</div>}
-  </div>
-);
+    /* eslint-enable no-nested-ternary */
```
Comment (harshithmohan) [inline src/pages/dashboard/panels/CollectionStats.tsx:45]:
```suggestion
```

File: src/pages/dashboard/panels/CollectionStats.tsx [hunk 3]
```diff
@@ -1,23 +1,50 @@
...
+  });
+
+  return (
+    /* eslint-disable no-nested-ternary */
+    <div className="flex">
+      <div className="grow">
+        {title}
+      </div>
+      {link
+        ? <Link to={link} className="font-semibold text-panel-text-primary">{value}</Link>
```
Comment (harshithmohan) [inline src/pages/dashboard/panels/CollectionStats.tsx:33]:
```suggestion
    <div className="flex">
      <div className="grow">
        {title}
      </div>
      {/* eslint-disable-next-line no-nested-ternary */}
      {link
        ? <Link to={link} className="font-semibold text-panel-text-primary">{value}</Link>
```

File: src/pages/collection/Collection.tsx [hunk 4]
```diff
@@ -232,7 +232,7 @@ function Collection() {
           className={cx(
             'flex items-start transition-all',
             (!isSeries && showFilterSidebar)
-              ? 'w-[28.84rem] opacity-100 overflow-auto '
+              ? 'w-[28rem] opacity-100 '
```
Comment (harshithmohan) [inline src/pages/collection/Collection.tsx:235]:
```suggestion
              ? 'w-[28rem] opacity-100'
```

File: src/components/Collection/TimelineSidebar.tsx [hunk 5]
```diff
@@ -37,22 +38,23 @@ const TimelineItem = ({ series }: { series: SeriesType }) => {
...
+        <div className="flex grow items-center justify-center text-panel-text-primary">
+          <Icon path={mdiLoading} size={3} spin />
+        </div>
+      )
+      : (
+        <div className="flex flex-col gap-y-3">
+          {series.map(item => <TimelineItem series={item} key={item.IDs.ID} />)}
+        </div>
+      )}
+  </ShokoPanel>
```
Comment (harshithmohan) [inline src/components/Collection/TimelineSidebar.tsx:57]:
```suggestion
  <ShokoPanel
    title="Timeline"
    className="sticky top-24 ml-6 !h-[calc(100vh-18rem)] w-[26.5rem]"
    contentClassName="gap-y-3"
    isFetching={isFetching}
  >
    {series.map(item => <TimelineItem series={item} key={item.IDs.ID} />)}
  </ShokoPanel>
```

File: src/components/Collection/SeriesTopPanel.tsx [hunk 6]
```diff
@@ -25,24 +26,24 @@ const SeriesTag = React.memo(({ text, type }: { text: string, type: 'User' | 'An
...
 
   return (
-    <div
+    <Button
       className={cx(
         'text-sm font-semibold flex gap-x-3 items-center border-2 border-panel-tags rounded-lg py-2 px-3 whitespace-nowrap capitalize h-fit cursor-pointer',
         type === 'User' ? 'text-panel-icon-important' : 'text-panel-icon-action',
       )}
       onClick={handleClick}
+      tooltip="Filter Tag"
```
Comment (harshithmohan) [inline src/components/Collection/SeriesTopPanel.tsx:42]: Do we need the tooltip?
It looks bad

File: src/components/Collection/ListViewItem.tsx [hunk 7]
```diff
@@ -45,17 +48,30 @@ const renderFileSources = (sources: SeriesSizesFileSourcesType): string => {
...
+  });
+
+  return (
+    <Button
+      className={cx(
+        'pointer-events-auto text-xs font-semibold flex gap-x-2 items-center border-2 border-panel-tags rounded-lg p-2 whitespace-nowrap capitalize cursor-pointer',
+        type === 'User' ? 'text-panel-text-important' : 'text-panel-text-primary',
+      )}
+      onClick={handleClick}
+      tooltip="Filter Tag"
```
Comment (harshithmohan) [inline src/components/Collection/ListViewItem.tsx:68]: See the comment on series top panel tooltip

File: src/components/Collection/Filter/Criteria.tsx [hunk 8]
```diff
@@ -58,8 +61,14 @@ const Criteria = ({ criteria, parameterExists, transformedParameter, type }: Pro
...
-    if (parameterExists) return;
-    setShowModal(true);
+    if (parameterExists) {
+      navigate('/webui/collection', { replace: true, state: { isTagLink: false } });
+      return;
+    }
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+    if (!location.state?.isTagLink) setShowModal(true);
+    // locate/navigate are only used to check/clear the tag link state, adding it to the deps would cause them to loop
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (harshithmohan) [inline src/components/Collection/Filter/Criteria.tsx:71]: Revert this, look at the comment where the filter is set

File: src/components/Collection/ListViewItem.tsx [hunk 9]
```diff
@@ -45,17 +48,30 @@ const renderFileSources = (sources: SeriesSizesFileSourcesType): string => {
...
-  </div>
-);
+const SeriesTag = React.memo(({ text, type }: { text: string, type: 'User' | 'AniDB' }) => {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const handleClick = useEventCallback(() => {
+    navigate('/webui/collection', { state: { isTagLink: true } });
+    addFilterCriteriaToStore('HasTag').then(() => {
+      dispatch(setFilterTag({ HasTag: [{ Name: text, isExcluded: false }] }));
+    }).catch(console.error);
```
Comment (harshithmohan) [inline src/components/Collection/ListViewItem.tsx:58]:
```suggestion
    dispatch(setFilterTag({ HasTag: [{ Name: text, isExcluded: false }] }));
    addFilterCriteriaToStore('HasTag').catch(console.error);
```

This will fix the modal issue. We don't need to wait for the criteria data to add the condition

File: src/components/Collection/SeriesTopPanel.tsx [hunk 10]
```diff
@@ -25,24 +26,24 @@ const SeriesTag = React.memo(({ text, type }: { text: string, type: 'User' | 'An
   const dispatch = useDispatch();
   const navigate = useNavigate();
   const handleClick = useEventCallback(() => {
-    dispatch(resetFilter());
```
Comment (harshithmohan) [inline src/components/Collection/SeriesTopPanel.tsx:28]: Why is this removed?
This will cause the tag condition to be added to the existing filter

File: src/components/Collection/SeriesTopPanel.tsx [hunk 11]
```diff
@@ -25,24 +26,24 @@ const SeriesTag = React.memo(({ text, type }: { text: string, type: 'User' | 'An
...
     addFilterCriteriaToStore('HasTag').then(() => {
       dispatch(setFilterTag({ HasTag: [{ Name: text, isExcluded: false }] }));
-      navigate('/webui/collection');
+      navigate('/webui/collection', { state: { isTagLink: true } });
```
Comment (harshithmohan) [inline src/components/Collection/SeriesTopPanel.tsx:31]: Doesn't need the state, look at the comment in ListViewItem

**Review by harshithmohan — CHANGES_REQUESTED** [review]

File: src/pages/dashboard/panels/CollectionStats.tsx (hunk 2 above)
Comment (harshithmohan) [reply src/pages/dashboard/panels/CollectionStats.tsx:45]: It will make sense when you look at the other suggestion I made for the code at the top 😔

File: src/components/Collection/SeriesTopPanel.tsx (hunk 6 above)
Comment (natyusha) [reply src/components/Collection/SeriesTopPanel.tsx:42]: 👍 Swapped to text hover instead

File: src/components/Collection/SeriesTopPanel.tsx (hunk 10 above)
Comment (natyusha) [reply src/components/Collection/SeriesTopPanel.tsx:28]: I expect some frustration if people have like 5 filters set up and it removes them all but it is linking from another page in this case so I'm not sure what is better (I put it back though)

File: src/components/Collection/TimelineSidebar.tsx [hunk 12]
```diff
@@ -17,42 +16,36 @@ const TimelineItem = ({ series }: { series: SeriesType }) => {
     : series.AniDB?.Type;
 
   return (
-    <Link to={`/webui/collection/series/${series.IDs.ID}`}>
-      <div className="flex gap-x-3" key={series.IDs.ID}>
+    <div className="flex gap-x-3" key={series.IDs.ID}>
+      <Link to={`/webui/collection/series/${series.IDs.ID}`}>
```
Comment (harshithmohan) [inline src/components/Collection/TimelineSidebar.tsx:20]: This would cause the title to not have a link..? 🤔

**Review by harshithmohan — CHANGES_REQUESTED** [review]

File: src/components/Collection/TimelineSidebar.tsx (hunk 12 above)
Comment (natyusha) [reply src/components/Collection/TimelineSidebar.tsx:20]: Yep, the same as the current collection list view so that the whitespace isn't a clickable link and for more easily selectable text. Adding a separate link to the title sounds good though so I did that and applied it to the list view too. For consistency's sake, I added similar behaviour to the poster view so that hovering over whitespace won't activate the poster zoom.
